src/AlgebraicGeometry/Schemes/CoveredSchemes/Objects/Methods.jl

```diff
@@ -387,3 +387,17 @@ function _compute_gluing(
     check=false
   )
 end
+
+@doc raw"""
+    irreducible_components(X::AbsCoveredScheme) -> Vector{Tuple{CoveredScheme,CoveredClosedEmbedding}}
+    
+Return the irreducible components of `X` with their reduced structure. 
```

Comment:
I don't mean to be nasty, but: If this is for schemes and not for varieties, is it OK then to ignore embedded components? Because from what I recall, the `maximal_associated_points` were dropping these. 

@afkafkafk13 : Any opinions? I think you had some more background knowledge with this?

Reply:
The notion of irreducible component is coming from (Zariski) topology.
https://stacks.math.columbia.edu/tag/004U
 Therefore I would argue that the embedded components have to be discarded. 
If we want embedded primes, then `associated_points` is the name to go and here we return ideal_sheave (I am O.K. with points being represented by ideal sheaves)

Reply:
The answers overlapped indeed. I agree with your suggestion.